libraries/platforms/dromajo-vcs/software/src/argp/argp-parse.c

```diff
@@ -0,0 +1,992 @@
+#  define gettext(msgid) (msgid)
+# endif
+#endif
+#ifndef N_
+# define N_(msgid) (msgid)
+#endif
+
+#if _LIBC - 0
+#include <bits/libc-lock.h>
+#else
+#ifdef HAVE_CTHREADS_H
+#include <cthreads.h>
+#endif
+#endif /* _LIBC */
+
+#include <argp.h>
+#include <argp-namefrob.h>
+
+/* Getopt return values.  */
+#define KEY_END (-1)		/* The end of the options.  */
+#define KEY_ARG 1		/* A non-option argument.  */
+#define KEY_ERR '?'		/* An error parsing the options.  */
+
+/* The meta-argument used to prevent any further arguments being interpreted
+   as options.  */
+#define QUOTE "--"
+
+/* The number of bits we steal in a long-option value for our own use.  */
+#define GROUP_BITS CHAR_BIT
+
+/* The number of bits available for the user value.  */
+#define USER_BITS ((sizeof ((struct option *)0)->val * CHAR_BIT) - GROUP_BITS)
+#define USER_MASK ((1 << USER_BITS) - 1)
+
+/* EZ alias for ARGP_ERR_UNKNOWN.  */
+#define EBADKEY ARGP_ERR_UNKNOWN
+
+/* Default options.  */
+
+/* When argp is given the --HANG switch, _ARGP_HANG is set and argp will sleep
+   for one second intervals, decrementing _ARGP_HANG until it's zero.  Thus
+   you can force the program to continue by attaching a debugger and setting
+   it to 0 yourself.  */
+volatile int _argp_hang;
+
+#define OPT_PROGNAME	-2
+#define OPT_USAGE	-3
+#define OPT_HANG	-4
+
+static const struct argp_option argp_default_options[] =
+{
+  {"help",	  '?',    	0, 0,  N_("Give this help list"), -1},
+  {"usage",	  OPT_USAGE,	0, 0,  N_("Give a short usage message")},
+  {"program-name",OPT_PROGNAME,"NAME", OPTION_HIDDEN, N_("Set the program name")},
+  {"HANG",	  OPT_HANG,    "SECS", OPTION_ARG_OPTIONAL | OPTION_HIDDEN,
+     N_("Hang for SECS seconds (default 3600)")},
+  {0, 0}
+};
+
+static error_t
+argp_default_parser (int key, char *arg, struct argp_state *state)
+{
+  switch (key)
+  {
+    case '?':
+      __argp_state_help (state, state->out_stream, ARGP_HELP_STD_HELP);
+    break;
+    case OPT_USAGE:
+      __argp_state_help (state, state->out_stream,
+          ARGP_HELP_USAGE | ARGP_HELP_EXIT_OK);
+    break;
+
+    case OPT_PROGNAME:		/* Set the program name.  */
+      program_invocation_name = arg;
+
+    /* [Note that some systems only have PROGRAM_INVOCATION_SHORT_NAME (aka
+    __PROGNAME), in which case, PROGRAM_INVOCATION_NAME is just defined
+    to be that, so we have to be a bit careful here.]  */
+    arg = strrchr (arg, '/');
+    if (arg)
+      program_invocation_short_name = arg + 1;
+    else
+      program_invocation_short_name = program_invocation_name;
+
+    /* Update what we use for messages.  */
+    state->name = program_invocation_short_name;
+
+    if ((state->flags & (ARGP_PARSE_ARGV0 | ARGP_NO_ERRS))
+        == ARGP_PARSE_ARGV0)
+      /* Update what getopt uses too.  */
+      state->argv[0] = program_invocation_name;
+
+    break;
+
+    case OPT_HANG:
+      _argp_hang = atoi (arg ? arg : "3600");
+      while (_argp_hang-- > 0)
+        // Sripathi:
```

Comment:
I replaced a call to sleep() with a for loop here since there does not seem to be a newlib implementation for it. Let me know what you guys think.